Source/Fuse.Common/Internal/Bitmap.uno

```diff
@@ -0,0 +1,501 @@
+		extern(Android) protected Bitmap(Java.Object nativeBitmap)
+		{
+			NativeBitmap = nativeBitmap;
+			_size = AndroidHelpers.GetSize(nativeBitmap);
+		}
+
+		extern(iOS) public readonly IntPtr NativeImage;
+		extern(iOS) protected Bitmap(IntPtr nativeImage)
+		{
+			NativeImage = nativeImage;
+			_size = IOSHelpers.GetSize(nativeImage);
+		}
+
+		extern(CIL) public readonly System.Drawing.Bitmap NativeBitmap;
+		extern(CIL) protected Bitmap(System.Drawing.Bitmap nativeBitmap)
+		{
+			NativeBitmap = nativeBitmap;
+			_size = new int2(nativeBitmap.Width, nativeBitmap.Height);
+		}
+
+		extern(!Android && !iOS && CPLUSPLUS) internal readonly CPlusPlusHelpers.NativeBitmapHandle NativeBitmap;
+		extern(!Android && !iOS && CPLUSPLUS) protected Bitmap(CPlusPlusHelpers.NativeBitmapHandle nativeBitmap)
+		{
+			NativeBitmap = nativeBitmap;
+			_size = CPlusPlusHelpers.GetSize(nativeBitmap);
+		}
+
+		static Bitmap LoadFromBundleFile(BundleFile bundleFile)
+		{
+			if (bundleFile.BundlePath.EndsWith(".gif"))
+			{
+				// TODO: Fuse.Diagnostic.InternalError
+				debug_log "Gif files are not fully supported!";
+			}
+
+			debug_log "File path:" + bundleFile.BundlePath;
+
+			if defined(CIL)
+				return LoadFromStream(bundleFile.OpenRead());
+			else if defined(Android)
+			{
+				var nativeBitmap = AndroidHelpers.DecodeFromBundle(bundleFile.BundlePath);
+				return new Bitmap(nativeBitmap);
+			}
+			else if defined(iOS)
+			{
+				var nativeImage = IOSHelpers.CreateImageFromBundlePath("data/" + bundleFile.BundlePath);
+				return new Bitmap(nativeImage);
+			}
+			else if defined(CPLUSPLUS)
+			{
+				var stream = (CppXliStream)bundleFile.OpenRead();
+				var nativeBitmap = CPlusPlusHelpers.LoadFromXliStream(bundleFile.Name, stream);
+				return new Bitmap(nativeBitmap);
+			}
+			else
+				build_error;
+		}
+
+		extern(CPLUSPLUS) static Bitmap LoadFromByteArray(string pathHint, byte[] data)
+		{
+			if defined(Android)
+			{
+				var byteBuffer = Android.Base.Wrappers.JWrapper.Wrap(Android.Base.Types.ByteBuffer.NewDirectByteBuffer(data));
+				var nativeBitmap = AndroidHelpers.DecodeFromByteBuffer(byteBuffer);
+				return new Bitmap(nativeBitmap);
+			}
+			else if defined(iOS)
+			{
+				var nativeImage = IOSHelpers.CreateImageFromByteArray(data);
+				return new Bitmap(nativeImage);
+			}
+			else
+			{
+				var nativeBitmap = CPlusPlusHelpers.LoadFromByteArray(pathHint, data);
+				return new Bitmap(nativeBitmap);
+			}
+		}
+
+		static extern(CIL) Bitmap LoadFromStream(Stream stream)
+		{
+			var nativeBitmap = new System.Drawing.Bitmap(stream);
+			return new Bitmap(nativeBitmap);
+		}
+
+		public static Bitmap LoadFromFileSource(FileSource fileSource)
+		{
+			var bundleFileSource = fileSource as BundleFileSource;
+			if (bundleFileSource != null)
+				return LoadFromBundleFile(bundleFileSource.BundleFile);
+
+			debug_log "File source" + fileSource.Name;
+			debug_log "Inside the loader..";
```

Comment:
@kusma This is still WIP, I committed to move debug traces between machines so I can test on another platform (force-pushed with the `debug_log`s gone away)